### PR TITLE
arch/sim/cmake: remove the host specific -U when HOSTSRCS. remove the bzero implement, use macro to memset.

### PR DESCRIPTION
**Pull request by jasonbu**
## Summary
fix macos compile hostfs.c compile issue.
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.0.sdk/usr/include/_string.h:131:62: error: expected function body after function declarator
  131 | char    *stpncpy(char *__dst, const char *__src, size_t __n) __OSX_AVAILABLE_STARTING(__MAC_10_7, __IPHONE_4_3);
      |                                     

when build native macos sim, memset zero will be automatic optimized to bzero. which caused 
memset->bzero->memset... deadloop, will abort on bootup.
as it's not a critical API, we should use a macro to map bzero to memset directly.

## Impact
fix macos cmake build problem.
bzero now is a macro mapped to memset, no function implement.

## Testing
CI-test & macos, sim:nsh Makefile&cmake local build.
```bash
# nuttx dir
cmake -B ../build -DBOARD_CONFIG=sim:nsh -GNinja
cmake --build ../build -j
../build/nuttx
```

### CODE REVIEW

File: arch/sim/src/sim/CMakeLists.txt [hunk 1]
```diff
@@ -288,4 +288,11 @@ endforeach()
 set(HOSTSRCS ${WINHOSTSRCS})
 
 target_sources(nuttx PRIVATE ${HOSTSRCS})
+
+get_target_property(HOST_COMPILE_OPTIONS nuttx COMPILE_OPTIONS)
+foreach(remove_item IN LISTS SIM_NO_HOST_OPTIONS)
+  list(REMOVE_ITEM HOST_COMPILE_OPTIONS ${remove_item})
+endforeach()
+set_target_properties(nuttx PROPERTIES COMPILE_OPTIONS
+                                       "${HOST_COMPILE_OPTIONS}")
```
Comment (xuxin930) [inline arch/sim/src/sim/CMakeLists.txt:297]:
```suggestion
nuttx_remove_compile_options(${SIM_NO_HOST_OPTIONS})
```

Comment (jasonbu) [reply arch/sim/src/sim/CMakeLists.txt:297]: nuttx_remove_compile_options not able to solve my problem, can you provide some sample how to make it work?

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by simbit18** [comment]
@xiaoxiang781216 @xuxin930  @jasonbu this pr broke build msvc

sched.vcxproj -> D:\a\nuttx_test_pr\nuttx_test_pr\sources\nuttx\vs2022\sched\Debug\sched.lib
  Generating nuttx_all.lib
  Microsoft (R) Library Manager Version 14.41.34123.0
  Copyright (C) Microsoft Corporation.  All rights reserved.
  
  Building Custom Rule D:/a/nuttx_test_pr/nuttx_test_pr/sources/nuttx/CMakeLists.txt
  Building Custom Rule D:/a/nuttx_test_pr/nuttx_test_pr/sources/nuttx/CMakeLists.txt
  HOST_COMPILE_OPTIONS-NOTFOUND
/HOST_COMPILE_OPTIONS-NOTFOUND(1,1): error C1083: Cannot open source file: 'HOST_COMPILE_OPTIONS-NOTFOUND': No such file or directory [D:\a\nuttx_test_pr\nuttx_test_pr\sources\nuttx\vs2022\nuttx.vcxproj]
  (compiling source file '/HOST_COMPILE_OPTIONS-NOTFOUND')

**Comment by jasonbu** [comment]
> @xiaoxiang781216 @xuxin930 @jasonbu this pr broke build msvc
> 
> https://github.com/simbit18/nuttx_test_pr/actions/runs/11498059450/job/32003163671#step:8:1780 sched.vcxproj -> D:\a\nuttx_test_pr\nuttx_test_pr\sources\nuttx\vs2022\sched\Debug\sched.lib Generating nuttx_all.lib Microsoft (R) Library Manager Version 14.41.34123.0 Copyright (C) Microsoft Corporation. All rights reserved.
> 
> Building Custom Rule D:/a/nuttx_test_pr/nuttx_test_pr/sources/nuttx/CMakeLists.txt Building Custom Rule D:/a/nuttx_test_pr/nuttx_test_pr/sources/nuttx/CMakeLists.txt HOST_COMPILE_OPTIONS-NOTFOUND /HOST_COMPILE_OPTIONS-NOTFOUND(1,1): error C1083: Cannot open source file: 'HOST_COMPILE_OPTIONS-NOTFOUND': No such file or directory [D:\a\nuttx_test_pr\nuttx_test_pr\sources\nuttx\vs2022\nuttx.vcxproj] (compiling source file '/HOST_COMPILE_OPTIONS-NOTFOUND')

@simbit18  I did not try the win host and re-produce the problem, please help me verify if it will work, I will do verify later.
please review #14499.
the problem should be COMPILE_OPTIONS is not set yet in nuttx cmake target at this time.

**Pull request merged**